src/composite.c

```diff
@@ -2,32 +2,42 @@
 #include <vips/vips.h>
 
 
+// Constants
 const int ALPHA_BAND_INDEX = 3;
 const int NUM_COLOR_BANDS = 3;
 
+// TODO: Copied from `common.cc`. Deduplicate once this becomes a C++ module.
+int HasAlpha(VipsImage *image) {
+  return (
+    (image->Bands == 2 && image->Type == VIPS_INTERPRETATION_B_W) ||
+    (image->Bands == 4 && image->Type != VIPS_INTERPRETATION_CMYK) ||
+    (image->Bands == 5 && image->Type == VIPS_INTERPRETATION_CMYK)
+  );
+}
```

Comment:
I realize it’s less convenient to have my code be straight C as I can’t reuse your function (as far as I know). Using C wasn’t a strong choice, but simply what I was slightly more familiar with, due to John’s VIPS examples.